### ISSUE TITLE
consul_service resource overrides node meta

### ISSUE THREAD

**Issue by ashald** [terraform-providers/terraform-provider-consul#101]
It's clear from the source code that we can see at https://github.com/terraform-providers/terraform-provider-consul/blob/master/consul/resource_consul_service.go#L321-L327:
```go
	var nodeMeta map[string]string
	nodeMeta = make(map[string]string)
	if d.Get("external").(bool) {
		nodeMeta["external-node"] = "true"
		nodeMeta["external-probe"] = "true"
	}
	registration.NodeMeta = nodeMeta
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `consul_service`

### Expected Behavior
At the bare minimum, one would expect that node meta would be preserved.

On top of that, I believe `external-node` and `external-probe` should not be overridden if already defined. For instance, we need to disable the ESM probe while keeping ESM health checks active so we set `external-probe = false` on the node.

**Comment by remilapeyre** [comment]
Thanks for the report @ashald, I overlooked this when adding `external-node` and `external-probe`. What you propose makes sense.

**Comment by ashald** [comment]
I'm pretty sure I'd do the same if I were in your shoes implementing those resources. There is a lot of complexity to it and it's easy to be tricked by that.

Thanks for your prompt attention!

**Comment by remilapeyre** [comment]
Thanks for your feedback, after more look into it, it seems that `consul_node` was not detecting changes made to the `address` and `meta` attributes which led to the bug in the first place. This is fixed in #104 that should be merged in the next few days.

In the meantime, you can work around the bug by removing `external` from your Terraform scenario and set `external-probe` and `external-node` manually in Consul node. It will be working as well as it was before (set the meta information but won't detect changes to it).

Once #104 is merged, it will start detecting external changes without any new action from your part.

**Comment by ashald** [comment]
Thanks for the update!
